palette/src/js/script.js

```diff
@@ -0,0 +1,409 @@
+class Palette {
+  constructor() {
+    this.mode = 'pensil';
+    this.currentColor = '#000000';
+    this.prevColor = '#ffffff';
+    this.permitDraw = false;
+    this.canvas = null;
+    this.ctx = null;
+    this.scale = 32;
+    this.firstPoint = {
+      left: null,
+      top: null,
+    };
+    this.busy = false;
+  }
+
+  // body load an other events
+  bodyLoad() {
+    this.createCanvas();
+    if (localStorage.getItem('state')) {
+      const state = JSON.parse(localStorage.getItem('state'));
+      this.mode = state.mode;
+      this.currentColor = state.currentColor;
+      this.prevColor = state.prevColor;
+      this.ctx.fillStyle = this.currentColor;
+      document.getElementById('inp-color').value = this.currentColor;
+    }
+    this.toggleTool(document.querySelector(`#${this.mode}`));
+    document.querySelector('#current-color i').style.color = this.currentColor;
+    document.querySelector('#prev-color i').style.color = this.prevColor;
+    document.querySelector('.left-menu').addEventListener('mousedown', (event) => { this.toggleTool(event); });
+    document.querySelector('.left-bottom-menu').addEventListener('mousedown', (event) => { this.toggleColor(event); });
+    document.getElementById('inp-color').addEventListener('change', (event) => { this.inputColor(event); });
+    // canvas events
+    this.canvas.addEventListener('mousedown', (event) => {
+      if (this.mode === 'pensil') {
+        this.startDraw(event);
+      } else if (this.mode === 'fill') {
+        if (!this.busy) {
+          this.busy = true;
+          this.fillPromise(event);
+        }
+      } else if (this.mode === 'choose-color') {
```

Comment:
It would be better to define tools as constants
e.g. `const tools = { pencil: 'choose-color' }`
because once you changed the string you have to change all the occurrences as well